target/linux/ath79/image/generic.mk

```diff
@@ -538,22 +538,23 @@ TARGET_DEVICES += glinet_gl-ar150
 define Device/glinet_gl-ar300m-common-nor
   ATH_SOC := qca9531
   DEVICE_VENDOR := GL.iNet
-  DEVICE_MODEL := GL-AR300M
   DEVICE_PACKAGES := kmod-usb-core kmod-usb2
   IMAGE_SIZE := 16000k
   SUPPORTED_DEVICES += gl-ar300m
 endef
 
 define Device/glinet_gl-ar300m-lite
   $(Device/glinet_gl-ar300m-common-nor)
+  DEVICE_MODEL := GL-AR300M
   DEVICE_VARIANT := Lite
 endef
 TARGET_DEVICES += glinet_gl-ar300m-lite
 
-define Device/glinet_gl-ar300m-nor
+define Device/glinet_gl-ar300m16
   $(Device/glinet_gl-ar300m-common-nor)
+  DEVICE_MODEL := GL-AR300M16
 endef
-TARGET_DEVICES += glinet_gl-ar300m-nor
+TARGET_DEVICES += glinet_gl-ar300m16
```

Comment:
This patch is really confusing me.
You just used the name gl-ar300m-nor in nand subtarget, and now you fix the double name by changing the device name in generic subtarget?
IMO, you should move this patch up front, so you do not have the same name twice.
In general, I would change the order in this patchset in a way that first everything is changed on generic target, and only then nand subtarget is touched (removing old nand in the beginning is okay ...).

Despite, with the changes as they are here, a user coming from ar71xx generic gl-ar300m-nor will update to ath79 nand gl-ar300m-nor?

Reply:
All upgrade paths checked months ago, clearly documented above at the start of this, as well as in the commit messages.

> Legacy NOR boards may be transitioned to full support of NAND without serial, U-Boot access, or use of "force" in sysupgrade. For example
>
> Legacy NOR ==> glinet,gl-ar300m-nor ==> glinet,gl-ar300m-nand

```
    Existing OpenWrt and GL.iNet firmware can be upgraded from NOR
    versions to glinet,gl-ar300m-nor and from NAND versions to
    glinet,gl-ar300m-nand without --force:
    
      (NOR)  gl-ar300m ==> glinet,gl-ar300m-nor ==> glinet,gl-ar300m-nand
      (NAND) gl-ar300m ==> glinet,gl-ar300m-nand
    
      glinet,gl-ar300m-nor <=> glinet,gl-ar300m-nand
    
    Upgrades are not possible using sysupgrade from legacy NOR to NAND
```

Reply:
> You just used the name gl-ar300m-nor in nand subtarget, and now you fix the double name by changing the device name in generic subtarget?

There's no double-naming. the gl-ar300m-nor was _moved_ from the generic target to the NAND target. Subsequently, the gl-ar30m16 was added. 

What you are seeing is the combination of diffs across multiple commits and how the diff algorithm chooses to determine what was removed or added. It's an illusion. Just like "Patches can be submitted as a pull request on Github"